config.go

```diff
@@ -116,20 +154,78 @@ func PrivateKey(key *rsa.PrivateKey) Option {
 	}
 }
 
+// PrivateKeyFile sets the RSA private key in the secure channel configuration
+// from a PEM or DER encoded file.
+func PrivateKeyFile(keyFile string) Option {
+	return func(c *uasc.Config, sc *uasc.SessionConfig) {
+		if keyFile == "" {
+			return
+		}
+
+		b, err := ioutil.ReadFile(keyFile)
+		if err != nil {
+			log.Fatalf("Failed to load private key: %s", err)
+		}
+
+		derBytes := b
+		if strings.HasSuffix(keyFile, ".pem") {
+			block, _ := pem.Decode(b)
+			if block == nil || block.Type != "RSA PRIVATE KEY" {
+				log.Fatalf("Failed to decode PEM block with private key")
+			}
+			derBytes = block.Bytes
+		}
+
+		pk, err := x509.ParsePKCS1PrivateKey(derBytes)
+		if err != nil {
+			log.Fatalf("Failed to parse private key: %s", err)
+		}
+
+		c.LocalKey = pk
+	}
+}
+
 // Certificate sets the client X509 certificate in the secure channel configuration
 // and also detects and sets the ApplicationURI from the URI within the certificate.
 func Certificate(cert []byte) Option {
 	return func(c *uasc.Config, sc *uasc.SessionConfig) {
-		c.Certificate = cert
+		setCertificate(cert, c, sc)
+	}
+}
+
+func setCertificate(cert []byte, c *uasc.Config, sc *uasc.SessionConfig) {
+	c.Certificate = cert
+
+	// Extract the application URI from the certificate.
+	var appURI string
+	x509cert, err := x509.ParseCertificate(cert)
+	if err == nil && len(x509cert.URIs) > 0 {
+		appURI = x509cert.URIs[0].String()
+	}
+	sc.ClientDescription.ApplicationURI = appURI
+}
+
+// X509KeyPair sets the client X509 certificate and private key in the secure
+// channel configuration from a pair of files and also detects and sets the
+// ApplicationURI from the URI within the certificate.
+func X509KeyPair(certFile, keyFile string) Option {
```

Comment:
If we add a `CertificateFile` (or similar) method, this now feels redundant and too constrained (these files needs to be PEM encoded, whereas the other methods would accept DER as well).  
I think it would be beneficial to have a generic `CertificateFile` method that accepts multiple file types, and if we wanted a convenience function that accepts both the cert and the key file simultaneously, it should just be a wrapper for the other two.  Both files are required so it makes sense to have a single method to load them.

Reply:
makes sense. Will do.

Reply:
Done.

Reply:
Makes way more sense this way.